src/snippets.md

```diff
@@ -0,0 +1,294 @@
+    this.instanceProp = value;
+  }
+
+  add(a: i32, b: i32): i32 {
+    return a + b;
+  }
+}
+
+class ExtendedClass extends BaseClass {
+
+  extendedProp: i32;
+
+  constructor(extendedValue: i32) {
+    super(1);
+
+    this.extendedProp = extendedValue;
+  }
+
+  add(a: i32): i32 {
+    return super.add(a, this.extendedProp + super.instanceProp);
+  }
+}
+
+export function getStaticProp(): i32 {
+  return ExtendedClass.staticProp;
+}
+
+export function overloadAdd(value: i32): i32 {
+  let extendedClass = new ExtendedClass(value);
+  return extendedClass.add(24);
+}
+
+#!html
+<script>
+const log = console.log;
+console.log = (...args) => {
+  log(...args);
+  let str = '';
+  args.forEach(arg => {
+    if (typeof arg == 'object') {
+      str += `${JSON.stringify(arg, null, 2)}<br/>`;
+    } else {
+      str += `${arg}<br/>`;
+    }
+  });
+  document.body.innerHTML += `<div>Log: ${str}</div>`;
+}
+
+loader.instantiate(module_wasm, { /* imports */ })
+.then(({ exports }) => {
+  console.log(`getStaticProp: ${exports.getStaticProp()}`);
+  console.log(`overloadAdd: ${exports.overloadAdd(24)}`);
+})
+</script>
+```
+
+## Handling Null
+
+Handling Null as a union type for optional return values, and simulating try/catch for errors in AssemblyScript
+
+```editor
+#!runtime=full
+class MyValue {
+  value: i32;
+
+  constructor(value: i32) {
+    this.value = value;
+  }
+}
+
+// Using a class type here, as some types are not nullable
+function getMyValue(isAble: boolean): MyValue | null {
+  let myValue = new MyValue(24);
+  if (isAble) {
+    return myValue;
+  } else {
+    return null;
+  }
+}
+
+export function positiveAddWithMyValue(a: i32): i32 {
+  let myValue = getMyValue(a > 0);
+  if (myValue == null) {
+    return -1;
+  } else {
+    return a + myValue.value;
+  }
+}
+
+#!html
+<script>
+const log = console.log;
+console.log = (...args) => {
+  log(...args);
+  let str = '';
+  args.forEach(arg => {
+    if (typeof arg == 'object') {
+      str += `${JSON.stringify(arg, null, 2)}<br/>`;
+    } else {
+      str += `${arg}<br/>`;
+    }
+  });
+  document.body.innerHTML += `<div>Log: ${str}</div>`;
+}
+
+loader.instantiate(module_wasm, { /* imports */ })
+.then(({ exports }) => {
+  if(exports.positiveAddWithMyValue(24) > -1) {
+    console.log("Add was successful")
+  } else {
+    console.log("Could not add 24");
+  }
+
+  if(exports.positiveAddWithMyValue(-1) > -1) {
+  console.log("Add was successful")
+  } else {
+  console.log("Could not add -1");
+  }
+});
+</script>
+```
+
+## Switch
+
+Using `switch` statements in AssemblyScript.
+
+**NOTE:** Switch conditions currently implicitly convert to u32, i.e. switching over strings or similar is not yet supported. If using i32s, a more technical detail would be that using values greater than or equal to zero has better codegen, because br_tables are unsigned, so a -1 for example would not tablify well with otherwise positive values (it does work, though, but code is not optimal).
```

Comment:
```suggestion
**NOTE:** Currently, the switch conditions (case values) are implicitly converted to u32, i.e. switching over strings or similar is not yet supported.
```

I don't understand this part 🙄, and I am not sure if these details are necessary here:
```
 If using i32s in the switch conditions, a more technical detail would be that using values greater than or equal to zero has better codegen, because br_tables are unsigned, so a -1, for example, would not tablify well with otherwise positive values (it does work, though, but code is not optimal).
```

Reply:
Hmm, I suggested this in https://github.com/AssemblyScript/website/pull/2#discussion_r436265041, but if you say it's not helping then I'm fine with keeping this simple.